tools/toolchains/arm.py

```diff
@@ -56,6 +56,11 @@ def __init__(self, target, options=None, notify=None, macros=None, silent=False,
         else:
             cpu = target.core
 
+        if not TOOLCHAIN_PATHS['ARM']:
+            exe = find_executable('armcc')
+            if exe:
```

Comment:
I did not raise an exception because I did nog want the constructor to fail. The exception will be raised in either compile_sources, link_program, or build_library